src/detect.c

```diff
@@ -1174,6 +1174,12 @@ static bool DetectRunTxInspectRule(ThreadVars *tv,
             inspect_flags, total_matches, engine);
 
     if (engine == NULL && total_matches) {
+        if (stored_flags) {
+            if (DetectEnginePktInspectionRun(tv, det_ctx, s, f, p, NULL) == false) {
```

Comment:
nit: Don't compare with bool values ... use `!DetectEnginePktInspectionRun`

Shouldn't `stored_flags` be updated with the value of `inspect_flags`?

Reply:
> Shouldn't stored_flags be updated with the value of inspect_flags?

It gets done a few lines below 
```
    if (stored_flags) {
        *stored_flags = inspect_flags;
```
Right ?